Reaction Dataset Socket does not support ReactionDatasetEntryStoichiometry class

**Describe the bug**
When adding an entry to the reaction dataset type, the `ReactionDatasetNewEntry` class takes an input with types: `stoichiometries: List[Union[ReactionDatasetEntryStoichiometry, Tuple[float, Union[int, Molecule]]]]`. However when providing the entry to the reaction dataset with `ds.add_entry` the following error occurs:
```
  File "bin/QCFractal/qcfractal/qcfractal/components/reaction/dataset_socket.py", line 53, in <listcomp>
    molecules = [x[1] for x in entry.stoichiometries]

TypeError: 'ReactionDatasetEntryStoichiometry' object is not subscriptable
```
and
```
E             File "bin/QCFractal/qcfractal/qcfractal/components/reaction/dataset_socket.py", line 62, in _create_entries
E               coefficient=coeff[0],
E                           ~~~~~^^^
E           TypeError: 'ReactionDatasetEntryStoichiometry' object is not subscriptable
E            (HTTP status 500)
```

**To Reproduce**
From tests in #910 
```
from qcfractal.components.reaction import testing_helpers

snowflake_client = snowflake.client()
storage_socket = snowflake.get_storage_socket()
manager_name, _ = snowflake.activate_manager()

ds = snowflake_client.add_dataset("reaction", "Test Reaction Dataset")
input_spec, stoichiometries, _ = testing_helpers.load_test_data("rxn_H2O_psi4_b3lyp_sp")
testing_helpers.run_test_data(storage_socket, manager_name, "rxn_H2O_psi4_b3lyp_sp")
 ds.add_entry(name="test_molecule", stoichiometries=stoichiometries)
```

**Expected behavior**
I expect no errors.

--- a/qcfractal/qcfractal/components/reaction/dataset_socket.py
+++ b/qcfractal/qcfractal/components/reaction/dataset_socket.py
@@ -50,7 +50,7 @@ class ReactionDatasetSocket(BaseDatasetSocket):
         all_entries = []
         for entry in new_entries:
             # stoichiometries = list of tuples
-            molecules = [x[1] for x in entry.stoichiometries]
+            molecules = [x[1] if isinstance(x, tuple) else x.molecule for x in entry.stoichiometries]
 
             meta, mol_ids = self.root_socket.molecules.add_mixed(molecules, session=session)
 
@@ -59,7 +59,7 @@ class ReactionDatasetSocket(BaseDatasetSocket):
             for coeff, mid in zip(entry.stoichiometries, mol_ids):
                 new_stoich_orm.append(
                     ReactionDatasetStoichiometryORM(
-                        coefficient=coeff[0],
+                        coefficient=coeff[0] if isinstance(coeff, tuple) else coeff.coefficient,
                         molecule_id=mid,
                     )
                 )

